src/sentry/monitors/endpoints/organization_monitors.py

```diff
@@ -77,17 +77,31 @@ def get(self, request: Request, organization: Organization) -> Response:
         else:
             environments = list(Environment.objects.filter(organization_id=organization.id))
 
-        # sort monitors by top monitor environment
-        monitorenvironment_top_status = Subquery(
-            MonitorEnvironment.objects.filter(
-                monitor__id=OuterRef("id"), environment__in=environments
+        # sort monitors by top monitor environment, then by last check-in
+        monitor_environments_query = MonitorEnvironment.objects.filter(
+            monitor__id=OuterRef("id"), environment__in=environments
+        )
+
+        queryset = queryset.annotate(
+            environment_status_ordering=Case(
+                When(status=MonitorStatus.DISABLED, then=Value(4)),
```

Comment:
Hardcoding the value 4 here is brittle, instead what we probably want to do is something like
```suggestion
                When(status=MonitorStatus.DISABLED, then=Value(len(DEFAULT_ORDERING) + 1)),
```

Reply:
We'll also want to change this to `ObjectStatus` probably

Reply:
yeah makes sense...I'd rather do all the status updates in one go

Reply:
Yeah sounds good